app/src/main/java/org/desarrolladorslp/teckersapp/ui/deliverables/DeliverableAdapter.kt

```diff
@@ -0,0 +1,106 @@
+package org.desarrolladorslp.teckersapp.ui.deliverables
+
+import android.os.Build
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RequiresApi
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.deliverable_item.view.*
+import org.desarrolladorslp.teckersapp.R
+import org.desarrolladorslp.teckersapp.model.DeliverableHeader
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+
+data class DeliverableAdapter(private val deliverablesHeader: ArrayList<DeliverableHeader>) :
+    RecyclerView.Adapter<DeliverableAdapter.DeliverableHeaderHolder>() {
+    private var deliverableCount =0
+
+    fun add(deliverableHeader: DeliverableHeader, position: Int) {
```

Comment:
You have the option here to give "position" a default value... I see that if it's -1, then you add it to the end. In my opinion, this should be the default behavior of `add`. You can do that like this:
```suggestion
    fun add(deliverableHeader: DeliverableHeader, position: Int = -1) {
```
this way, you can call `add(deliverable)` and it will be addet to the end of the list automatically..